### ISSUE TITLE
Include exotic keyboard layout variants

### ISSUE THREAD

**Issue by krumeluu** [calamares/calamares#1826]
When choosing a keyboard layout, only the most common variants are listed.
I'd like to be able to choose a more exotic variant, such as the ones listed in `.../xkb/rules/evdev.extras.xml`.

My current workaround is to choose the most similar variant from the English layouts and choose the proper layout only after install. This is "fine", but the layouts not being a complete match with my local variant, and not being able to see the password input makes this very slow and error-prone.

It's probably a good idea not to clutter the menu with all the exotic layout variants out there. So I would suggest a button/toggle 'show more layouts' or such, which would reveal/hide the exotic variants of the language chosen.

**Comment by adriaandegroot** [comment]
- A PR to lift the keyboard-layout-drawing code from the KDE Keyboards KCM (I think it shows one) would be welcome. That's a pretty big replacement of the widget, but would also partly solve another issue in Calamares, about showing dead-keys.

- A PR to read that XML file and add its entries to the keyboard models list would be welcome. The files are read when initializing the global keyboard model. The code is in `src/modules/keyboard/keyboardwidget/keyboardglobal.cpp`. It could use some refactoring to make eg. `parseKeyboardLayouts()` place its data into a map-passed-by-reference; then adding a `parseXMLKeyboardLayouts()` can do the same and together they populate the whole map which is then returned from `KeyboardGlobal::getKeyboardLayouts()`. It might require a bit of extra logic in the sorting-logic for display, though, to separate the exotic from the ordinary (or a `QSortFilterProxyModel`, leaving the existing sorting intact).

**Comment by adriaandegroot** [comment]
Feedback timeout.
